sw/device/tests/crypto/cryptotest/firmware/extclk_sca_fi.c

```diff
@@ -0,0 +1,76 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/tests/crypto/cryptotest/firmware/extclk_sca_fi.h"
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/tests/crypto/cryptotest/firmware/status.h"
+#include "sw/device/tests/crypto/cryptotest/json/extclk_sca_fi_commands.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_clkmgr_t clkmgr;
+
+/**
+ * Extclk Configure command handler.
+ *
+ * Allows to configure the external clock.
+ * The uJSON data contains:
+ *  - sel: External clock on or off.
+ *  - hi_speed_sel: Nominal or low-speed external clock.
+ *
+ * @param uj The received uJSON data.
+ */
+status_t handle_extclk_sca_fi_configure(ujson_t *uj) {
+  cryptotest_extclk_sca_fi_cfg_t uj_data;
+  TRY(ujson_deserialize_cryptotest_extclk_sca_fi_cfg_t(uj, &uj_data));
+
+  UJSON_CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  bool is_low_speed = true;
+  if (uj_data.hi_speed_sel == 1) {
+    is_low_speed = false;
+  }
+
+  if (uj_data.sel == 0) {
+    // Disable external clock.
+    UJSON_CHECK_DIF_OK(dif_clkmgr_external_clock_set_disabled(&clkmgr));
```

Comment:
Not sure where these `UJSON_CHECK*` macros came from, but why can you not just use the `TRY()` macro here? it seems like `UJSON_CHECK_DIF_OK()` masks the status produced and returns an `ABORTED()`? Is there a reason for this?

Reply:
Yeah, you are right - fixed.